### PR TITLE
follow up PR for test_c10d_ucc.py in response to Xiang's review of #88110

### PR DESCRIPTION
**Pull request by Fuzzkatt**
* Adds extra test_allgather_base in UccProcessGroupWithDispatchedCollectivesTests; rest of nccl and gloo tests there don't work on ucc
* Adds cpu tests for [op]_work_wait_gpu tests
* Added single tensor input test for allgather_basics; multi tensor input still doesn't seem to be supported by ucc

### CODE REVIEW

File: test/distributed/test_c10d_ucc.py [hunk 1]
```diff
@@ -1031,19 +1072,61 @@ def test_nested_comm_tensor_wrapping_gpu(self):
...
+    @skip_if_lt_x_gpu(1)
+    def test_allgather_base(self):
+        store = dist.FileStore(self.file_name, self.world_size)
+        dist.init_process_group(
+            "ucc",
+            world_size=self.world_size,
+            rank=self.rank,
+            store=store,
+        )
+        device = "cuda"
```
Comment (kwen2501) [inline test/distributed/test_c10d_ucc.py:1124]: There is no device ID here. Is this based on an assumption that `set_current_device` has been called somewhere before?

Comment (Fuzzkatt) [reply test/distributed/test_c10d_ucc.py:1124]: nccl equivalent of this test uses default cuda device (https://github.com/pytorch/pytorch/blob/main/test/distributed/test_c10d_nccl.py#L2995), so I just followed the convention. Doesn't seem like cuda.set_device is used.

Comment (kwen2501) [reply test/distributed/test_c10d_ucc.py:1124]: So this test just tests for correct dispatching to device backend.
And the base test class assumes a world_size of 1:
https://github.com/pytorch/pytorch/blob/main/test/distributed/test_c10d_common.py#L1692-L1695

**Review by kwen2501 — APPROVED** [review]

**Pull request closed without merging**